server/app/report.go

```diff
@@ -224,3 +231,42 @@ func (a *UserSurveyApp) generateSurveyMetadataFile(survey *model.Survey, key str
 
 	return filePath, nil
 }
+
+func (a *UserSurveyApp) generateServerMetadataFile(survey *model.Survey, key string) (string, error) {
+	metadata, err := a.generateServerMetadata()
+	if err != nil {
+		return "", errors.Wrap(err, "generateServerMetadataFile: failed to generate server metadata")
+	}
+
+	filePath := path.Join(os.TempDir(), "survey_report", key, "server_metadata.yaml")
+	file, err := os.Create(filePath)
+	if err != nil {
+		a.api.LogError("generateServerMetadataFile: failed to create server metadata YAML file", "surveyID", survey.ID, "key", key, "filePath", filePath, "error", err.Error())
+		return "", errors.Wrapf(err, "generateServerMetadataFile: failed to create server metadata YAML file, surveyID: %s, key: %s, filePath: %s", survey.ID, key, filePath)
+	}
+
+	defer file.Close()
+
+	yamlEncoder := yaml.NewEncoder(file)
+	err = yamlEncoder.Encode(metadata)
+	if err != nil {
+		a.api.LogError("generateServerMetadataFile: failed to encode server metadata via YAML encoder", "surveyID", survey.ID, "key", key, "filePath", filePath, "error", err.Error())
+		return "", errors.Wrapf(err, "generateServerMetadataFile: failed to encode server metadata via YAML encoder, surveyID: %s, key: %s, filePath: %s", survey.ID, key, filePath)
+	}
+
+	return filePath, nil
+}
+
+func (a *UserSurveyApp) generateServerMetadata() (*mmModel.Metadata, error) {
+	manifest := a.manifest
```

Comment:
@harshilsharma63 , as mentioned in another comment, it'd be good to align to Ibrahim's metadata package -see Metadata structure in https://github.com/mattermost/mattermost/pull/27465/files.

Reply:
https://github.com/mattermost/mattermost/pull/27490 will ensure that all plugins use the same format.

Reply:
@catalintomai I didn't understand your comment, but I'm guessing its fixed after using @hanzei 's changes?